presto-orc/src/main/java/com/facebook/presto/orc/OrcRecordReader.java

```diff
@@ -321,22 +324,31 @@ public long getMaxCombinedBytesPerRow()
     public void close()
             throws IOException
     {
-        orcDataSource.close();
-
-        if (writeChecksumBuilder.isPresent()) {
-            WriteChecksum actualChecksum = writeChecksumBuilder.get().build();
-            validateWrite(validation -> validation.getChecksum().getTotalRowCount() == actualChecksum.getTotalRowCount(), "Invalid row count");
-            List<Long> columnHashes = actualChecksum.getColumnHashes();
-            for (int i = 0; i < columnHashes.size(); i++) {
-                int columnIndex = i;
-                validateWrite(validation -> validation.getChecksum().getColumnHashes().get(columnIndex).equals(columnHashes.get(columnIndex)),
-                        "Invalid checksum for column %s", columnIndex);
+        try (Closer closer = Closer.create()) {
```

Comment:
I would write this as

```
try (Closer closer = Closer.create()) {
    closer.register(orcDataSource);
    for (StreamReader column : streamReaders) {
        if (column != null) {
            closer.register(() -> column.close());
        }
    }
}
```

I made two changes here:
1) `orcDataSource` is closeable so I can just register the object itself.
2) Registered the individual columns in a for loop to make sure that all of them get closed. With the current code the for loop lambda will be called and if a column.close() throws the rest will not get closed.

Reply:
oh thanks for the correction!